src/app/component/add-product/add-product.component.ts

```diff
@@ -20,18 +21,35 @@ import { AppState } from 'src/app/store/state/app.state';
 export class AddProductComponent implements OnInit, OnDestroy {
   productAddedSubscription$ = new Subscription();
 
+  productForm = this.fb.group({
+    name: ['', Validators.required],
+    category: ['', Validators.required],
+    image: ['', Validators.required],
+    price: ['', Validators.required],
+    description: ['', Validators.required],
+  });
+
   constructor(
     private store: Store<AppState>,
     private location: Location,
+    private fb: FormBuilder,
     private actionsSubject$: ActionsSubject
   ) {}
 
   ngOnInit(): void {
     this.productAddedSubscription$ = this.actionsSubject$
       .pipe(ofType<AddProductSuccess>(ProductActionTypes.AddProductSuccess))
       .subscribe(() => {
-        // TODO: use html element instead alert to notify the user
-        alert('product created');
+        this.store.dispatch(
```

Comment:
this will better be a side-effect of the AddProductSuccess instead of a subscription here. As a general rule you don't listen for the effects inside the components